electron-app/src/components/settings/api-keys/ExchangeSettings.vue

```diff
@@ -120,12 +132,39 @@ export default class ExchangeSettings extends Vue {
   showPassphrase: boolean = false;
 
   connectedExchanges!: string[];
+  krakenAccountTypes = ['starter', 'intermediate', 'pro'];
+  selectedKrakenAccountType = '';
+
+  mounted() {
+    this.selectedKrakenAccountType = this.settings.krakenAccountType;
+  }
 
   @Watch('selectedExchange')
-  onChange() {
+  onChangeExchange() {
     this.resetFields();
   }
 
+  @Watch('selectedKrakenAccountType')
+  onChangeKrakenAccountType() {
+    const { commit } = this.$store;
+    this.$api
```

Comment:
this would be better located in an action maybe https://github.com/rotki/rotki/blob/08562c304c528f2d2bf2f8a7556a33d0dac0194d/electron-app/src/store/session/actions.ts

Reply:
you can name the action `setKrakenAccountType` and then do a dispach from here

Reply:
Why should this be an action? What is the difference? What are the advantages? This change should be instantaneous.

One thing I really don't like is how this pops popup window saying success every time. What's better would be to just show a green checkmark next to the v-select meny to signify it's set.

I mean thinking about it this backend API call can't fail. You are setting only 1 setting which woud only fail if you provide an invalid value, but you don't.

Reply:
@LefterisJP I was thinking the same thing about the accounting settings to be honest I also don't like the popup for a single setting. Well the benefit mostly is that anything mutating the state is not part of the component. In general I am trying to reduce any api calls/state muntations in the components.

Reply:
But what I mean is I thought that making it into an action somehow makes it async? I  guess from your answer that it does not?

If so I will make an attempt but I may screw up the syntax.

Reply:
This is what we're talking about here right: https://github.com/rotki/rotki/issues/751 ?

Reply:
@isidorosp similar yes. That conversation was mostly about errors if I recall but it's the same thing.

Reply:
Well it is already async :P promises and `async functions` are basically the same. If you mean that

Reply:
No last night I meant the whole async way of querying the backend with ascyn_query = True .. but well it was late :)

Can't be like that if the backend call itself is not async. Doh! :P